### PR TITLE
Added List type to support subdocuments

### PR DESCRIPTION
**Pull request by glennvangastel**
This PR adds support for subdocuments in keystone

GUI needs more work, but is out of scope of my project

Example of how this should be implemented : 

```
employees: {type: Types.List, fields: {
    userID: {type: Types.Relationship, ref: 'User', initial: true},
    specialHours: {type: Types.List, fields: {
      start: {type: Types.Datetime},
      end: {type: Types.Datetime},
    }},
    specialDays: {type: Types.List, fields: {
      monday: {type: Types.Boolean},
      tuesday: {type: Types.Boolean},
      wednesday: {type: Types.Boolean},
      thursday: {type: Types.Boolean},
      friday: {type: Types.Boolean},
      saturday: {type: Types.Boolean},
      sunday: {type: Types.Boolean}
    }}
  }}
```

### CODE REVIEW

File: fields/types/list/ListColumn.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import ItemsTableCell from '../../components/ItemsTableCell';
+import ItemsTableValue from '../../components/ItemsTableValue';
+import { plural } from '../../../admin/client/utils/string';
+
+var ListColumn = React.createClass({
```
Comment (knor-el-snor) [inline fields/types/list/ListColumn.js]: Babel included in package.json, so normally ES6 syntax will be supported.

> export default class ListColumn extends React.Component {
> ...
> }

Remove module.exports

File: fields/types/list/ListColumn.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	displayName: 'ListColumn',
+	propTypes: {
+		col: React.PropTypes.object,
+		data: React.PropTypes.object,
+	},
+	getValue () {
+		var value = this.props.data.fields[this.props.col.path];
+		if (Array.isArray(value)) {
```
Comment (knor-el-snor) [inline fields/types/list/ListColumn.js]: Shorthand method:

> Array.isArray(value) ? plural(value.length,'\* Value', '*Values') : "";

Just a personal preference

File: fields/types/list/ListColumn.js [hunk 3]
```diff
@@ -3,30 +3,26 @@ import ItemsTableCell from '../../components/ItemsTableCell';
...
+export default class ListColumn extends React.Component {
+	displayName: 'ListColumn'
+	props: {
 		col: React.PropTypes.object,
 		data: React.PropTypes.object,
-	},
+	}
+
 	getValue () {
 		var value = this.props.data.fields[this.props.col.path];
```
Comment (hoetmaaiers) [inline fields/types/list/ListColumn.js]: if ES6 available, never use `var`!

Comment (glennvangastel) [reply fields/types/list/ListColumn.js]: Missed that one, thanks!

**Pull request merged**